fast_llm/layers/ssm/config.py

```diff
@@ -0,0 +1,138 @@
+from fast_llm.tensor import TensorSpace
+from fast_llm.utils import Assert
+
+
+class SSMDimNames:
+    d_model = "model_dimension_D"
+    d_state = "state_dimension_N"
+    d_conv = "size_of_conv1d_input"  # dimention of the conv1d input in mamba layers
+    d_inner = "inner_dimension_after_expansion"
+    dt_rank = "rank_of_Δ"
+    d_inner_proj_m = "inner_projection_dimension_mamba"
+    d_inner_proj_m2 = "inner_projection_dimension_mamba2"
+    d_x_proj = "x_projection_dimension"
+    headdim = "head_dimension_P"  # dimention of the mamba2 head
+    d_conv_kernel = "1d_conv_kernel_size"  # kernel size of the conv1d in mamba layers
+    n_qk_heads = "number_of_qk_heads"
+    n_v_heads = "number_of_v_heads"
+
+
+@config_class()
+class SSMArchitectureConfig(BaseModelArchitectureConfig):
+    _abstract = False
+
+    # Normalization
+    normalization: NormalizationArchitectureConfig = Field(
+        default_factory=NormalizationArchitectureConfig,
+        desc="Configuration for the normalization layers architecture.",
+        hint=FieldHint.core,
+    )
+
+    expansion_factor: int = Field(
+        default=2, desc="Expansion factor for Mamba blocks.", hint=FieldHint.core, valid=check_field(Assert.gt, 0)
+    )
+
+    state_size: int = Field(
+        default=16,
+        desc="State size for Mamba blocks.",
+        hint=FieldHint.core,
+        valid=check_field(Assert.gt, 0),
+    )
+    conv_kernel_dimension: int = Field(
+        default=4,
+        desc="Conv kernel dimension for Mamba blocks.",
+        hint=FieldHint.core,
+        valid=check_field(Assert.gt, 0),
+    )
+
+    # Layer parameters
+    add_bias_linear: bool = Field(
+        default=False,
+        desc="Whether to use bias in SSM layers",
+        hint=FieldHint.core,
+    )
+
+    dt_rank: str | int = Field(
+        default="auto",
+        desc="Rank of the Δ projection matrix. If 'auto', set to ceil(hidden_size/16)",
+        hint=FieldHint.core,
+    )
+
+    chunk_size: int = Field(
+        default=256,
+        desc="Chunk size for Mamba2 blocks.",
+        hint=FieldHint.core,
+    )
+
+    n_qk_heads: int = Field(
+        default=32,
+        desc="Number of QK heads for Mamba2 blocks.",
+        hint=FieldHint.core,
+    )
+
+    n_v_heads: int = Field(
+        default=32,
+        desc="Number of V heads for Mamba2 blocks.",
+        hint=FieldHint.core,
+    )
+
+    activation: str = Field(
+        default="silu",
+        desc="Activation function for Mamba2 blocks.",
+        hint=FieldHint.core,
+    )
+
+    dt_min: float = Field(
+        default=0.001,
+        desc="Minimum step size for discretization",
+        hint=FieldHint.core,
+        valid=check_field(Assert.gt, 0),
+    )
+
+    dt_max: float = Field(
+        default=0.1,
+        desc="Maximum step size for discretization",
+        hint=FieldHint.core,
+        valid=check_field(Assert.gt, 0),
+    )
+
+    dt_init_floor: float = Field(
```

Comment:
Init args are non-architecture.

Reply:
@jlamypoirier the architecture/non-architecture distinction is non-intuitive and leaks an implementation detail of config resolution into the rest of the codebase. I don't see why @oleksost or anybody else should have to care about it. We need to remove such obstacles. Let's work on it.